articles/115_topic_and_service_name_mapping.md

```diff
@@ -190,25 +182,35 @@ Here is a table with some example expansions:
 
 Note the private namespace substitution character makes the name absolute, and therefore the namespace is not added a second time.
 
+### Substitutions
+
+The bracket syntax (`{substitution_name}`) may be used in non-fully qualified names to substitute useful contextual information into the name.
+The set of substitution keys (names) are not set in this document, but some reasonable examples might be: `{node}` expands to the current node's name or `{ns}` expands to the current node's namespace.
+
+Substitutions are expanded before the private namespace substitution character is expanded.
+Therefore a substitution may contain the private namespace substitution character, i.e. `~`.
+For example, given the name `{private}foo`, if there is a substitution called `{private}` that expands to `~/_`, the current node name is `my_node`, and the current node's namespace is `my_ns`, then the fully qualified name would be `/my_ns/my_node/_foo`.
+
 ### Hidden Topic or Service Names
 
-Any topic or service name that contains any tokens (either namespaces or a topic or service name) which starts with an underscore (`_`) will be considered hidden and tools may not show them unless explicitly asked.
+Any topic or service name that contains any tokens (either namespaces or a topic or service name) that start with an underscore (`_`) will be considered hidden and tools may not show them unless explicitly asked to.
```

Comment:
I would prefer it if the sentence did not end with a preposition. Though that seems to be an outdated tradition: http://grammar.yourdictionary.com/parts-of-speech/prepositions/Ending-a-Sentence-with-a-Preposition.html